src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2362,38 +2361,82 @@ public override BoundNode VisitConditionalAccess(BoundConditionalAccess node)
 
         public override BoundNode VisitConditionalOperator(BoundConditionalOperator node)
         {
-            var isByRef = node.IsRef;
-
             VisitCondition(node.Condition);
             var consequenceState = this.StateWhenTrue;
             var alternativeState = this.StateWhenFalse;
 
+            bool consequenceEndReachable;
+            bool alternativeEndReachable;
+            TypeWithState consequenceRValue;
+            TypeWithState alternativeRValue;
+
+            if (node.IsRef)
+            {
+                TypeWithAnnotations consequenceLValue;
+                TypeWithAnnotations alternativeLValue;
+
+                (consequenceLValue, consequenceRValue) = visitConditionalRefOperand(consequenceState, node.Consequence);
+                consequenceState = this.State;
+                consequenceEndReachable = this.State.Reachable;
+                (alternativeLValue, alternativeRValue) = visitConditionalRefOperand(alternativeState, node.Alternative);
+                alternativeEndReachable = this.State.Reachable;
+                Join(ref this.State, ref consequenceState);
+
+                TypeSymbol refResultType = node.Type.SetUnknownNullabilityForReferenceTypes();
+                if (IsNullabilityMismatch(consequenceLValue, alternativeLValue))
+                {
+                    // l-value types must match
+                    ReportNullabilityMismatchInAssignment(node.Syntax, consequenceLValue, alternativeLValue);
+                }
+                else if (!node.HasErrors)
+                {
+                    refResultType = consequenceRValue.Type.MergeNullability(alternativeRValue.Type, VarianceKind.None);
+                }
+
+                var lValueAnnotation = (consequenceEndReachable, alternativeEndReachable) switch
```

Comment:
>lValueAnnotation [](start = 20, length = 16)

I'm sorry to be a pain about this, but I don't believe computing a nullable annotation that differs based on reachability is correct.

Reply:
My bad, I thought you objected to doing it for an unreachable branch, but I hadn't understood you meant for the whole ternary as well.

Pushed a commit for that and filed https://github.com/dotnet/roslyn/issues/34411 to confirm language design. Thanks